aiven/client/client.py

```diff
@@ -400,6 +400,47 @@ def get_service_topic(self, project, service, topic):
         path = self.build_path("project", project, "service", service, "topic", topic)
         return self.verify(self.get, path, result_key="topic")
 
+    def list_m3_namespaces(self, project, service):
+        service_res = self.get_service(project=project, service=service)
+        return service_res.get("user_config", {}).get("namespaces", [])
+
+    def delete_m3_namespace(self, project, service, ns_name):
+        service_res = self.get_service(project=project, service=service)
+        old_namespaces = service_res.get("user_config", {}).get("namespaces", [])
+        new_namespaces = [ns for ns in old_namespaces if ns["name"] != ns_name]
+        if len(old_namespaces) == len(new_namespaces):
+            raise KeyError(f"Namespace '{ns_name}' does not exist")
```

Comment:
Should rather print a meaningful error than raising exception which is then visible to the users as a stacktrace.

Reply:
Fixed in cli.py by catching this and raising args.UserError, which seems to print out nicely. This seems to be done in cli.py for other errors.